zrcs/zrc-7.md

```diff
@@ -0,0 +1,238 @@
+
+## IV. Specification
+
+The metadata specification describes:
+
+1. the metadata interface that is returned by an API for developers to implement and ecosystems to consume
+2. the localization interface for developers to implement and ecosystems to consume
+3. the dynamic interpolation that may be present in the URL string, such as {id} and {locale}
+
+### A. Rationale
+
+Compared to ERC-1155, the JSON Metadata schema allows to indicate the MIME type of the files pointed by any URI field. This is to allow clients to display appropriately the resource without having to first query it to find out the MIME type.
+
+Metadata localization should be standardized to increase presentation uniformity across all languages. As such, a simple overlay method is proposed to enable localization.
+
+NFT's are broken into two parts. The first is the blockchain smart contract, which contains the association between users and a token. Each token when examained points off-chain to an external resource which contains the metadata for that particular token. This could be on the surface web or using a decentralised storage solution such as IPFS or Arweave.
+
+ZRC-6 gives the developer the ability to set an ```base_uri``` which exposes a field for the developer to insert a URI they control. When navigating to ```base_uri/{id}``` for a given ```token_id```, the response returned should be a standardised JSON structure for it to be considered ZRC-7 compliant.
+
+The main advantages of keeping metadata offchain are as follows.
+
+* Mutability
+  * Owner can update the metadata at anytime without a chain call (changing hashes of IPFS and Arweave resources)
+* Storage
+  * Owner can store more data offchain than onchain and for a cheaper cost to maintain and amend.
+
+The main disadvantages of keeping metadata offchain are as follows.
+
+* Mutability 
+  * Owner can maliciously update the metadata at anytime without a chain call.
+
+### B. Metadata JSON Schema
+
+```js
+{
+    "name": "Cloud Quackers",
+    "description": "Non-Fungible Ducks",
+    "resource": "ipfs://QmTwmV66VMgq5YzEUZj48oqvznkiUGVMCT4UnWRjr1L9gU",
+    "resource_mimetype": "image/png",
+    "external_url": "https://duck.community",
+    "external_description": "Non-Fungible Ducks are the first randomised, hard-cap project on the Zilliqa blockchain.",
+    "attributes": 
+    [ 
+          {
+            "display_type" : "string",
+            "trait_type": "Base name", 
+            "value": "Mandarin Shiny"
+          }, 
+          {
+            "display_type" : "string",
+            "trait_type": "Base rarity percent", 
+            "value": "10%"
+          }, 
+          {
+            "display_type" : "string",
+            "trait_type": "Base occurance chance", 
+            "value": "0.91%"
+          }
+    ],
+    "localization": {
+        "type": "object",
+        "properties": {
+            "uri": "",
+            "default": "",
+            "locales": ""
+        }
+    }
```

Comment:
The schema from eip-1155 is represented in JSON and it is not a template. You misunderstood the JSON Metadata schema.

This is how it works.
```
{
  "name": "Advertising Space",
  "description": "Each token represents a unique Ad space in the city.",
  "localization": {
    "uri": "ipfs://QmWS1VAdMD353A6SDk9wNyvkT14kyCiZrNDYAad4w1tKqT/{locale}.json",
    "default": "en",
    "locales": ["en", "es", "fr"]
  }
}
```
base URI for the localization: `ipfs://QmWS1VAdMD353A6SDk9wNyvkT14kyCiZrNDYAad4w1tKqT/{locale}.json`

`ipfs://QmWS1VAdMD353A6SDk9wNyvkT14kyCiZrNDYAad4w1tKqT/es.json` should return:
```
{
  "name": "Espacio Publicitario",
  "description": "Cada token representa un espacio publicitario único en la ciudad."
}
```

`ipfs://QmWS1VAdMD353A6SDk9wNyvkT14kyCiZrNDYAad4w1tKqT/fr.json` should return:
```
{
  "name": "Espace Publicitaire",
  "description": "Chaque jeton représente un espace publicitaire unique dans la ville."
}
```